storage/innobase/buf/buf0buf.cc

```diff
@@ -3777,6 +3777,16 @@ dberr_t buf_page_t::read_complete(const fil_node_t &node)
   if (err == DB_PAGE_CORRUPTED || err == DB_DECRYPTION_FAILED)
   {
 release_page:
+    if (node.space->full_crc32() && node.space->crypt_data
+        && recv_recovery_is_on())
```

Comment:
Lines should be split after binary operators such as `&&`, not before them.

It looks like `recv_sys_t::recovery_on` had better be defined as `Atomic_relaxed<bool>`, and the assignment to `false` should be documented as being protected by `mutex`.

Not all assignments to `false` are currently protected by `recv_sys.mutex`. Before the call to `recv_recovery_read_max_checkpoint()` there is an assignment that is not protected. This assignment seems to be redundant, because `recv_recovery_from_checkpoint_start()` should not have been invoked yet.